core/src/com/unciv/logic/map/MapUnit.kt

```diff
@@ -678,7 +687,18 @@ class MapUnit {
         // getAncientRuinBonus, if it places a new unit, does too
         currentTile = tile
 
-        if (tile.improvement == Constants.ancientRuins && civInfo.isMajorCiv())
+        if (civInfo.isMajorCiv() 
+            && tile.improvement != null
+            // I don't know who came up with the idea of making starting locations improvements, 
+            // but I don't like having to make this check here.
+            // Why not just cache these somewhere for easy access? Why do they have to be saved
+            // on the map instead of in some variable? I can't imagine it being easier to
+            // find where they are this way, and this kind of random check feels way out of place.
+            // This is just misuse of an existing variable for a completely different usecase at this point.
+            // Now we have to check this every time a unit moves over an improvement, which happens _quite often_.
```

Comment:
The answer is that this needs to be in the TileMap somewhere, since when you save a map that's the data object that gets serialized.
I agree that it could be anywhere in the TileMap, I think saving it within a specific tile is the correct thing. Whether this should be an improvement or not... not sure about that.

Reply:
Downside of making it an improvement is that in random places you need a "!tile.improvement.startsWith("StartingLocation ")" check when this is only relevant when placing startingUnits. For now I'll leave it like this, but maybe sometime in the future I'll create a PR to fix this.

Reply:
I say - this definitely needs to go away from being a pseudo-improvement. A saved map looks like this:
`{mapParameters:{name:My new map,type:Custom,..},tileList:[{position:{x:29,y:32},baseTerrain:Grassland},...,{position:{x:-7,y:20},baseTerrain:Grassland}]}`
... so there _is_ room parallel to the tileList. I say a `class StartingPosition(val nation: String, val location: Vector2)` and a `val startingPositions = listOf<StartingPosition>()` in TileMap - ignored everywhere except game start, cleared there and for safety in save game... 'In a specific tile' is ... a misunderstanding. No such kludges necessary!